### PR TITLE
Show correct error message

### PR DESCRIPTION
**Pull request by dem4ron**
https://github.com/exercism/website/assets/66035744/8f641579-f6a1-4076-8c32-cefc03b9aa99

### CODE REVIEW

File: app/javascript/utils/fetch-json.ts [hunk 1]
```diff
@@ -1,35 +1,46 @@
...
 
-export const fetchJSON = <T extends any>(
+export class ApiError extends Error {
+  type: string
+
+  constructor(type: string, message: string) {
+    super(message)
+    this.name = this.constructor.name
+    this.type = type
+    Object.setPrototypeOf(this, ApiError.prototype)
```
Comment (ErikSchierboom) [inline app/javascript/utils/fetch-json.ts]: Why is this needed?

File: app/javascript/utils/fetch-json.ts [hunk 2]
```diff
@@ -1,35 +1,46 @@
 import { camelizeKeys } from 'humps'
 
-export const fetchJSON = <T extends any>(
+export class ApiError extends Error {
+  type: string
+
+  constructor(type: string, message: string) {
```
Comment (ErikSchierboom) [inline app/javascript/utils/fetch-json.ts]: Could this be:

```suggestion
  constructor(public type: string, message: string) {
```

File: app/javascript/utils/fetch-json.ts (hunk 1 above)
Comment (dem4ron) [reply app/javascript/utils/fetch-json.ts]: For Typescript. It is needed to correctly set the prototype chain when extending built-in classes. This ensures that instance methods, like instanceof, function correctly.

File: app/javascript/utils/fetch-json.ts (hunk 2 above)
Comment (dem4ron) [reply app/javascript/utils/fetch-json.ts]: yes!

File: db/schema.rb [hunk 3]
```diff
@@ -39,6 +39,11 @@
     t.index ["blob_id", "variation_digest"], name: "index_active_storage_variant_records_uniqueness", unique: true
   end
 
+  create_table "add_uuid_to_code_tags_samples", charset: "utf8mb4", collation: "utf8mb4_unicode_ci", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
```
Comment (ErikSchierboom) [inline db/schema.rb]: I think this should probably be removed, right @iHiD

File: app/views/training_data/dashboard/index.html.haml
Comment (ErikSchierboom) [inline app/views/training_data/dashboard/index.html.haml:1]: app/views/training_data/dashboard/index.html.haml was moved to app/views/training_data/code_tags_samples/index.html.haml. Could you update?

Comment (dem4ron) [reply app/views/training_data/dashboard/index.html.haml:1]: then I think the whole `app/views/training_data/dashboard` folder can be deleted, right?

Comment (dem4ron) [reply app/views/training_data/dashboard/index.html.haml:1]: ~also `code_tags_samples` could be removed, and its content moved one level up.~

File: app/controllers/training_data/dashboard_controller.rb
Comment (ErikSchierboom) [inline app/controllers/training_data/dashboard_controller.rb:1]: Could you revert this?
